src/random_nums_config.cpp

```diff
@@ -0,0 +1,64 @@
+#include "include/random_nums_config.hpp"
+#include "duckdb/common/string_util.hpp"
+#include "duckdb/common/local_file_system.hpp"
+
+#include <iostream>
+#include <fstream>
+
+namespace duckdb {
+
+using namespace duckdb_yyjson;
+
+RandomNumsConfig::RandomNumsConfig() {
+    GetDefaultConfig();
+}
+
+RandomNumsConfig::RandomNumsConfig(string &file_path) {
+    GetDefaultConfig();
+    GetConfigFromFile(file_path);
+}
+
+RandomNumsConfig::~RandomNumsConfig() {
+}
+
+void RandomNumsConfig::GetDefaultConfig() {
+    select_percentage = 60;
+    attach_percentage = 40;
+    attach_use_percentage = 50;
+    detach_percentage = 60;
+    set_percentage = 30;
+    delete_percentage = 40;
+}
+
+void RandomNumsConfig::GetConfigFromFile(string &file_path) {
+    // open and read file into a string &
+    string result;
+    std::ifstream open_file(file_path);
+    if (!open_file.is_open()) {
+        std::cerr << "Could not open the file!" << std::endl;
+    }
+    std::string line;
+    while (std::getline(open_file, line)) {
```

Comment:
Did you try any functions in the `yyjson` library?  I.e 
```
yyjson_api yyjson_doc *yyjson_read_file(const char *path,
                                        yyjson_read_flag flg,
                                        const yyjson_alc *alc,
                                        yyjson_read_err *err);
```

found in `third_party/yyjson/include/yyjson.hpp:873`

Edit: It's possible then that the `sql_smith` extension then requires the `json` extension, which is possible